File: SnowForecast/app/src/main/java/com/sners/snowforecast/data/DailyData.java
package com.sners.snowforecast.data;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.time.ZoneId;
import java.util.ArrayList;


public class DailyData extends IntervalData {

    private String date;
    private float tempMin;
    private float tempMax;


    ////////////////////////////////////////////////////////////////////////////////
    //	Constructor
    ////////////////////////////////////////////////////////////////////////////////
    public DailyData(JSONObject jsonDaily) {
        try {
            date = jsonDaily.getString(WeatherConstants.TIME);

            precipIntensity = (float) jsonDaily.getDouble(WeatherConstants.PRECIP_INTENSITY);
            precipProbability = (float) jsonDaily.getDouble(WeatherConstants.PRECIP_PROBABILITY);

            tempMin = (float) jsonDaily.getDouble(WeatherConstants.TEMP_MIN);
            tempMax = (float) jsonDaily.getDouble(WeatherConstants.TEMP_MAX);

            if (!jsonDaily.isNull(WeatherConstants.PRECIP_TYPE)) {
                //  If it is null, then it is not an array
                    JSONArray weatherWordsJson = jsonDaily.getJSONArray(WeatherConstants.PRECIP_TYPE);
                    for (int wordCounter = 0; wordCounter < weatherWordsJson.length(); wordCounter++) {
                        String precipType = weatherWordsJson.getString(wordCounter);
                        weatherWords.add(precipType);
                }
            }




        } catch (JSONException e) {
            e.printStackTrace();
        }

    }


    public String getDate() {
        return date;
    }

    public float getTempMin() {
        return tempMin;
    }

    public float getTempMax() {
        return tempMax;
    }
}

File: SnowForecast/app/src/main/java/com/sners/snowforecast/data/DailyData.kt
package com.sners.snowforecast.data

import org.json.JSONException
import org.json.JSONObject

/**
 * A data class of daily weather conditions
 * It takes the raw json and converts it into a data structure
 * @param jsonDaily The raw current data coming in from the API call
 */
class DailyData(jsonDaily: JSONObject) : IntervalData() {

    /**
     *  @property date Date as read from the json
     */
    var date: String? = null

    /**
     *  @property tempMin Minimum temperature in degrees C
     */
    var tempMin = 0f

    /**
     *  @property tempMax Maximum temperature in degrees C
     */
    var tempMax = 0f

    /*
  Constructor
   */
    init {
        try {
            date = jsonDaily.getString(WeatherConstants.TIME)
            precipIntensity = jsonDaily.getDouble(WeatherConstants.PRECIP_INTENSITY).toFloat()
            precipProbability = jsonDaily.getDouble(WeatherConstants.PRECIP_PROBABILITY).toFloat()
            tempMin = jsonDaily.getDouble(WeatherConstants.TEMP_MIN).toFloat()
            tempMax = jsonDaily.getDouble(WeatherConstants.TEMP_MAX).toFloat()
            if (!jsonDaily.isNull(WeatherConstants.PRECIP_TYPE)) {
                //  If it is null, then it is not an array
                val weatherWordsJson = jsonDaily.getJSONArray(WeatherConstants.PRECIP_TYPE)
                for (wordCounter in 0..weatherWordsJson.length() -1) {
                    val precipType = weatherWordsJson.getString(wordCounter)
                    weatherWords.add(precipType)
                }
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }
}